mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2388,32 +2546,445 @@ impl UtxoRpcClientOps for ElectrumClient {
+        }
+    }
+
+    fn on_disconnected(&self, address: String) {
+        info!("electrum_conn_mng on_disconnected from: {}", address);
+        let self_copy = self.clone();
+        self.0.abortable_system.weak_spawner().spawn(async move {
+            if let Err(err) = self_copy.clone().suspend_server(address.clone()).await {
+                error!("Failed to suspend server: {}, error: {}", address, err);
+            }
+            if let Err(err) = self_copy.connect().await {
+                error!(
+                    "Failed to reconnect after addr was disconnected: {}, error: {}",
+                    address, err
+                );
+            }
+        });
+    }
+}
+
+#[async_trait]
+impl ConnMngTrait for Arc<dyn ConnMngTrait + Send + Sync> {
+    async fn get_conn(&self, address: Option<&str>) -> Option<Arc<AsyncMutex<ElectrumConnection>>> {
+        self.deref().get_conn(address).await
+    }
+    async fn connect(&self) -> Result<(), String> { self.deref().connect().await }
+    async fn is_connected(&self) -> bool { self.deref().is_connected().await }
+    async fn remove_server(&self, address: String) -> Result<(), String> { self.deref().remove_server(address).await }
+    async fn register(&self, handler: RpcTransportEventHandlerShared) { self.deref().register(handler).await; }
+    async fn on_verified(&self, address: String) { self.deref().on_verified(address).await }
+    fn on_disconnected(&self, address: String) { self.deref().on_disconnected(address) }
+}
+
+impl ConnMngImpl {
+    fn new(
+        servers: Vec<ElectrumConnSettings>,
+        abortable_system: AbortableQueue,
+        event_handlers: Vec<RpcTransportEventHandlerShared>,
+    ) -> ConnMngImpl {
+        let mut primary: LinkedList<String> = LinkedList::new();
+        let mut backup: LinkedList<String> = LinkedList::new();
+        let mut abortable_systems: BTreeMap<String, ElectrumConnCtx> = BTreeMap::new();
+        for conn_settings in servers {
+            match conn_settings.priority {
+                Priority::Primary => primary.push_back(conn_settings.url.clone()),
+                Priority::Secondary => backup.push_back(conn_settings.url.clone()),
+            }
+            let _ = abortable_systems.insert(conn_settings.url.clone(), ElectrumConnCtx {
+                conn_settings,
+                connection: None,
+                abortable_system: abortable_system.create_subsystem().unwrap(),
+                suspend_timeout_sec: SUSPEND_TIMEOUT_INIT_SEC,
+            });
+        }
+
+        ConnMngImpl {
+            guarded: AsyncMutex::new(ConnMngState {
+                connecting: AtomicBool::new(false),
+                event_handlers,
+                queue: ConnMngQueue { primary, backup },
+                active: None,
+                conn_ctxs: abortable_systems,
+            }),
+            abortable_system,
+        }
+    }
+
+    pub async fn get_settings_to_connect(
```

Comment:
I think a better name would be `fetch_next_connection_settings`

Reply:
done